crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -369,18 +369,24 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
             ]) {
                 if let Expr::Attribute(ast::ExprAttribute { value, attr, .. }) = func.as_ref() {
                     let attr = attr.as_str();
-                    if let Expr::StringLiteral(ast::ExprStringLiteral { value: string, .. }) =
-                        value.as_ref()
+                    if let Expr::StringLiteral(ast::ExprStringLiteral {
+                        value: string_value,
+                        ..
+                    }) = value.as_ref()
                     {
                         if attr == "join" {
                             // "...".join(...) call
                             if checker.enabled(Rule::StaticJoinToFString) {
-                                flynt::rules::static_join_to_fstring(checker, expr, string);
+                                flynt::rules::static_join_to_fstring(
+                                    checker,
+                                    expr,
+                                    string_value.as_str(),
+                                );
                             }
```

Comment:
I think the previous code was easier to read than now where we have the explicit `as_str` call and using `Deref` seemed to work just fine?

Reply:
The motivation here is that the `Deref` implementation includes an allocation -- the conversion here can allocate if the string is implicitly concatenated. So `Deref` is essentially hiding an allocation. I prefer that we do something explicit over an implicit allocation. It's not about the lifetimes or anything like that. It's intentionally more verbose.

Reply:
Yeah, I think using `to_str` makes more sense. I was not aware of that. I'll create a follow-up PR to rename that.